scala/private/phases/phase_collect_jars.bzl

```diff
@@ -46,41 +41,32 @@ def phase_collect_jars_junit_test(ctx, p):
     )
     return _phase_collect_jars_default(ctx, p, args)
 
-def phase_collect_jars_library_for_plugin_bootstrapping(ctx, p):
-    args = struct(
-        unused_dependency_checker_mode = "off",
-    )
-    return _phase_collect_jars_default(ctx, p, args)
-
 def phase_collect_jars_common(ctx, p):
     return _phase_collect_jars_default(ctx, p)
 
 def _phase_collect_jars_default(ctx, p, _args = struct()):
     return _phase_collect_jars(
         ctx,
+        p.dependency,
```

Comment:
Would you mind to change it to just `p`? I know using `p.dependency` will be less verbose inside the function but just for the purpose of consistency.

Reply:
This change was in response to a comment by @ittaiz . I am fine either way.

Reply:
@borkaehw why do you think this is a better pattern in general? 
Let’s ignore consistency for a sec since maybe we’ll decide to change the pattern

Reply:
I think we should keep the function signature as simple as possible. By the fact that `p` should contain all the information we need, we may expand the data structure within the function instead.

In my opinion,
```
phase_a(ctx, p, a)
```
```
phase_b(ctx, p, a, b)
```
```
phase_c(ctx, p, a, b, c, d, e)
```
make the phase architecture messy. When we finalize #965, I imagine the functions should all be consistent like `phase_a(ctx, internal_proviers)`.

Also, functions with a super long list of arguments like `compile_scala` makes it hard to read. I, in general, prefer to keep arguments as less as possible and no keyword is needed.

Reply:
@borkaehw I understand the value in what you say but to be honest I actually quite disagree with it. I think that bottom line having similar APIs is less beneficial than having phases explicitly declare which internal providers they need. Something like `providers = [JavaInfo]` which we have in startlark.
Having said that I'd rather not start this debate here over this change and so for consistency I apologize @Jamie5 and I'll ask you to revert to your original change. Sorry again.

Reply:
Sounds good. Thanks for sharing your idea.